qadence/backends/horqrux/backend.py

```diff
@@ -135,18 +135,29 @@ def expectation(
             A jax.Array of shape (batch_size, n_observables)
         """
         observable = observable if isinstance(observable, list) else [observable]
-        batch_size = max([arr.size for arr in param_values.values()])
+        if "observables" in param_values or "circuit" in param_values:
+            raise NotImplementedError("The HORQRUX backend does not support separated parameters.")
+        else:
+            merged_params = param_values
+            batch_size = max([arr.size for arr in param_values.values()])  # type: ignore[union-attr]
         n_obs = len(observable)
 
-        def _expectation(params: ParamDictType) -> ArrayLike:
+        def _expectation(params: ParamDictType | SeparatedParamDictType) -> ArrayLike:
+            param_circuits = params["circuit"] if "circuit" in params else params
+            param_observables = params["observables"] if "observables" in params else params
             out_state = self.run(
-                circuit, params, state, endianness, horqify_state=True, unhorqify_state=False
+                circuit,
+                param_circuits,
+                state,
+                endianness,
+                horqify_state=True,
+                unhorqify_state=False,
             )
-            return jnp.array([o.native(out_state, params) for o in observable])
+            return jnp.array([o.native(out_state, param_observables) for o in observable])
```

Comment:
```suggestion
            return jnp.array([observable.native(out_state, param_observables) for observable in observables])
```

I guess we don't have to be stingy on names ;)